src/inference/AdvancedSMC.jl

```diff
@@ -99,44 +128,54 @@ Usage:
 PG(100, 100)
 ```
 """
-mutable struct PG{space, F} <: InferenceAlgorithm
+mutable struct PG{space, F} <: ParticleInference
   n_particles           ::    Int         # number of particles used
-  n_iters               ::    Int         # number of iterations
   resampler             ::    F           # function to resample
 end
-function PG(n_particles::Int, n_iters::Int, resampler::F, space::Tuple) where F
-    return PG{space, F}(n_particles, n_iters, resampler)
+function PG(n_particles::Int, resampler::F, space::Tuple) where F
+    return PG{space, F}(n_particles, resampler)
 end
-PG(n1::Int, n2::Int, ::Tuple{}) = PG(n1, n2)
-function PG(n1::Int, n2::Int, space::Symbol...)
-    PG(n1, n2, resample_systematic, space)
+PG(n1::Int, ::Tuple{}) = PG(n1)
+function PG(n1::Int, space::Symbol...)
+    PG(n1, resample_systematic, space)
 end
 
+alg_str(spl::Sampler{PG}) = "PG"
+
 const CSMC = PG # type alias of PG as Conditional SMC
 
-function Sampler(alg::PG, s::Selector)
+"""
+    Sampler(alg::PG, model::Model, s::Selector)
+
+Return a `Sampler` object for the PG algorithm.
+"""
+function Sampler(alg::T, model::Model, s::Selector) where T<:PG
     info = Dict{Symbol, Any}()
```

Comment:
Do we still need `info` after introducing `state`?

Reply:
Yeah, because `VarInfo` uses the `info` field. It'll need to be removed completely in a separate PR.